js/gc-smart-data.js

```diff
@@ -336,6 +336,9 @@ window.GC = window.GC || {};
 
         if (boneAgeList && boneAgeList.length) {
             $.each(boneAgeList, function(i, o) {
+                if (o.hasOwnProperty("boneAgeMos") && o.hasOwnProperty("date")) {
+                    o = new SmartBoneage(o.date, o.boneAgeMos);
+                }
                 if (o instanceof SmartBoneage) {
                     patient.boneAge.push(o.toGCBoneage(patient));
                 } else {
```

Comment:
In what scenario is the value in `patient.boneAge` in the else condition used?

Reply:
If its a valid Bone Age observation we should always have `boneAgeMos` and `date` available but incase if any of these are not available it should hit the `else` route. 

From my investigation for this issue, the app is currently looking for the `SmartBoneage` object but it didnt get it as the app was not creating it anywhere (hence this PR). 
We might be able to remove the `else` case here (since with this PR we expect SmartBoneage object to be created) but I haven't investigated where bone age is used in the calculations. Hence decided to retain the logic so that I dont break any calculations down the line accidentally.